README.md

```diff
@@ -100,6 +100,7 @@ More detailed instructions are in the [`example README.md`](https://github.com/R
 | **cwd** | string | no | Specify the current working directory for your container process. If the directory does not exist, one will be created for you. |
 | **privileged** | bool | no | Run container in privileged mode. Your container will have all linux capabilities when running in privileged mode. |
 | **pids_limit** | int64 | no | An integer value that specifies the pid limit for the container. Defaults to unlimited. |
+| **hostname** | string | no | The hostname to assign to the container. When launching more than one of a task (using `count`) with this option set, every container the task starts will have the same hostname. |
```

Comment:
So by `default` it does what you are suggesting:

1) If the user doesn't supply the `hostname` option at all. The default `hostname` assigned to the container will be the container name `{taskName}-{allocId}`. This is similar to `nomad docker driver` where the default `hostname` assigned is the `containerID`.

```
group "redis-group" {
   count = 2 
   task "redis-task" {
      driver = "containerd-driver"
       config {
           image   = "redis:alpine"
       }
}
```
This will launch 2 containers `c1` and `c2` with **unique** hostnames `redis-task-<c1's-alloc-id>` and `redis-task-<c2's-alloc-id>` respectively.

2) If the user supplies the `hostname` in the `TaskConfig` then we should honor what the user requested for 🙂 

```
group "redis-group" {
   task "redis-task" {
      driver = "containerd-driver"
       config {
           image   = "redis:alpine"
           hostname = "foobar"
       }
}
```
In the above job, 1 task (container) will be launched, with the hostname `foobar`. If the user is launching two different tasks (containers) and want unique hostnames for those 2 containers, they just need to supply two different `hostnames`.

3) If the user launches the same job as in (2) but with `count=2`, then `the hostname will be **same** for both the containers`, since the users requested for that hostname. **NOTE** This is same behavior as [`nomad docker driver`](https://www.nomadproject.io/docs/drivers/docker#hostname)

```
When launching more than one of a task (using count) with this option set, every container the task starts will have the same hostname.
```

Reply:
All concerns cleared up!